### ISSUE TITLE
SQM_reads.pl error multithread

### ISSUE THREAD

**Issue by 3ccasto** [jtamames/SqueezeMeta#220]
[error_read.txt](https://github.com/jtamames/SqueezeMeta/files/5753860/error_read.txt)
Hi,

I'm running the sqm_reads.pl of 69 samples with:

`$SQUEEZEDIR/utils/sqm_reads.pl -p READS -s samples_read.txt -f cleaning -t 21 -b15`

At the beginning the program gave me the following error:

```
Starting multithread LCA in 21 threads: 1 2 DBD::SQLite::db prepare failed: attempt to write a readonly database at /home/cricasre/SqueezeMeta/lib/SQM_reads/lca_reads.pl line 208, <inc> line 3.
Thread 1 terminated abnormally: DBD::SQLite::db prepare failed: attempt to write a readonly database at /home/cricasre/SqueezeMeta/lib/SQM_reads/lca_reads.pl line 208, <inc> line 3.
```

The program is running but in each samples  shows the same error. I attached a file with the complete error.

**Comment by fpusan** [comment]
Might be related to this

https://stackoverflow.com/questions/17793672/perl-dbi-treats-setting-sqlite-db-cache-size-as-a-write-operation-when-subclassi

Do you have write permissions in the database directory?

In any case, no write operations should be needed, so we might want to look deeper into that.

**Comment by 3ccasto** [comment]
The permissions of my database directory are the folowwing:
`rigel$ ls -ld db
drwxrwxr-x 10 cricasre gente 4096 Dec 29 05:57 db
`

**Comment by fpusan** [comment]
Just in case, can you also look at the permissions on `db/LCA_tax` and all the files within?

**Comment by fpusan** [comment]
While you are at it, let's try the following

1) Edit `/home/cricasre/SqueezeMeta/lib/SQM_reads/lca_reads.pl`
2) Change line 144 so it looks like this:
   ```my $dbh = DBI->connect("dbi:SQLite:dbname=$lca_db","","",{ RaiseError => 1, sqlite_open_flags => SQLITE_OPEN_READONLY, sqlite_use_immediate_transaction => 0}) or die $DBI::errstr;```

In any case LCA reads is working consistently in our test machines, so I suspect the issue is in your server configuration. Let's see what happens...

**Comment by 3ccasto** [comment]
The same error happens when I run the 06.lca.pl in the seqmerged mode. I think that could be an error in the configuration of perl modules. The permissions of db/LCA_tax are the following:
`rigel$ ls -lthr db/LCA_tax/
total 50G
-rw-r--r-- 1 cricasre gente 4.9K Dec 29 05:31 gencode.dmp
-rw-r--r-- 1 cricasre gente  452 Dec 29 05:31 division.dmp
-rw-r--r-- 1 cricasre gente 2.9K Dec 29 05:36 typeoftype.dmp
-rw-r--r-- 1 cricasre gente  19M Dec 29 05:36 typematerial.dmp
-rw-r--r-- 1 cricasre gente 258M Dec 29 05:36 taxidlineage.dmp
-rw-r--r-- 1 cricasre gente 284M Dec 29 05:36 rankedlineage.dmp
-rw-r--r-- 1 cricasre gente 1.1M Dec 29 05:36 merged.dmp
-rw-r--r-- 1 cricasre gente 610M Dec 29 05:36 fullnamelineage.dmp
-rw-r--r-- 1 cricasre gente 4.0M Dec 29 05:36 delnodes.dmp
-rw-r--r-- 1 cricasre gente 191M Dec 29 05:38 nodes.dmp
-rw-r--r-- 1 cricasre gente 183M Dec 29 05:38 names.dmp
-rw-r--r-- 1 cricasre gente  18M Dec 29 05:38 citations.dmp
-rw-r--r-- 1 cricasre gente 5.2M Dec 29 05:38 host.dmp
-rw-rw-r-- 1 cricasre gente 111M Dec 29 05:58 new_taxdump.tar.gz
-rw-rw-r-- 1 cricasre gente 289M Dec 29 05:58 taxatree.txt
-rw-rw-r-- 1 cricasre gente 299M Dec 29 06:01 parents.txt
-rw-rw-r-- 1 cricasre gente  12G Dec 29 08:25 nr.taxlist.db
-rw-rw-r-- 1 cricasre gente  35G Dec 29 10:15 taxid_tree.txt
-rw-r--r-- 1 cricasre gente 4.5K Dec 29 10:15 taxid.db-journal
-rw-r--r-- 1 cricasre gente 950M Dec 29 12:32 taxid.db
`

**Comment by fpusan** [comment]
I have similar permissions `-rw-r--r--` in my own machine, and it's working.
Also, the `06.lca.pl` script used to work for you, right?

**Comment by 3ccasto** [comment]
Yes, In the other modes. I have talked with the administrator of the server because the program run differently depend on the nodes where it runs. Could I execute the sqm_reads.pl for 69 samples in my our computer? My OS is Intel Core i7-8565 U CPU @ 1.80 GHz 1.99 GHz with RAM 16 Gb.

**Comment by fpusan** [comment]
Yes, provided you have enough disk space to store the databases AND results. Running 69 samples there will take a lot, though.
Maybe you can run only one sample, just to check that it finishes successfully.

**Comment by 3ccasto** [comment]
Ok, I will check

**Comment by 3ccasto** [comment]
I solved the problem. I installed again the databases using donwload_databases.pl. When I used the script make_databases.pl the program take a lot of time in the generation of the databases using SQLite. My job run during 24h of run, but It finished before the completely generation of databases. That was the error. However, the download of databases using make_databases.pl spend a lot of time for me, is that normal?
